overcooked_ai_py/mdp/overcooked_env.py

```diff
@@ -76,21 +76,20 @@ def __init__(self, mdp_generator_fn, start_state_fn=None, horizon=MAX_HORIZON, m
         self.mlp_params = mlp_params
         self.start_state_fn = start_state_fn
         self.info_level = info_level
-        # self.lst_len = 8
-        # self.mdp_lst = [self.mdp_generator_fn() for _ in range(self.lst_len)]
+        self.lst_len = 4
+        self.mdp_lst = [self.mdp_generator_fn() for _ in range(self.lst_len)]
```

Comment:
Maybe I'm forgetting the conclusions about our previous conversations about this, so correct me if I'm wrong. I thought that we concluded that in most cases (eventually) we would have environments with infinitely many mdps (case which I don't think is currently supported).

Because of that, it seems to me like the cleanest way is not to have the `self.mdp_lst` attribute, and just have `self.mdp_generator_fn()` return one of 4 mdps at random (or even simply iterate between them deterministically). Is there any case in which we explicitly need to have `self.mdp_lst`?

Reply:
You are right. On the side, I was also running experiment on the effect of outer-shape on the side, and I usually first contrain the number of MDPs when getting a grip of where the hyperparameters should be starting (If I set it to be infinite, it will take many many iterations before I can know if the hyperparameters is even within the ball-park) But I will update those to reflect the infinite version like we agreed on before.

Reply:
Yeah, mainly what i was saying was that even if you had a `self.mdp_generator_fn()` (that supports infinite # of MDPs), you could still represent the 1 (or few MDP case), by simply doing a variation of the following (in mostly pseudo-code):

```
mdps = [LayoutGenerator().generate_mdp(CUSTOM_SETTINGS) for _ in range(NUM_LAYOUTS)]
mdp_generator_fn()  = lambda: np.random.choice(mdps)
```

This should enable you to do testing of all kinds, if I understand correctly what you're trying to do.